src/App.tsx

```diff
@@ -1,74 +1,125 @@
-              </div>
+  const applyQuery = useCallback(
+    debounce((value: string) => setAppliedQuery(value), 300),
+    [],
+  );
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  const filteredList = useMemo(() => {
+    return appliedQuery
+      ? people.filter(person => person.name.includes(appliedQuery))
+      : people;
+  }, [appliedQuery, people]);
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsActive(true);
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setSelectedPerson(null);
+  };
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+  return (
+    <>
+      <div className="container">
+        <main className="section is-flex is-flex-direction-column">
+          <h1 className="title" data-cy="title">
+            {selectedPerson
+              ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+              : 'No selected person'}
+          </h1>
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
+          <div className="dropdown is-active">
+            <div className="dropdown-trigger">
+              <input
+                type="text"
+                placeholder="Enter a part of the name"
+                className="input"
+                data-cy="search-input"
+                value={query}
+                onChange={handleQueryChange}
+                onFocus={() => {
+                  setIsActive(true);
+                  if (!query) {
+                    setAppliedQuery('');
+                  }
+                }}
+              />
+            </div>
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
+            {isActive && filteredList.length > 0 && (
+              <div
+                className="dropdown-menu"
+                role="menu"
+                data-cy="suggestions-list"
+              >
+                <div className="dropdown-content">
+                  {filteredList.map(person => (
+                    <div
+                      className="dropdown-item"
+                      data-cy="suggestion-item"
+                      key={person.name}
+                      onClick={() => {
+                        setSelectedPerson(person);
+                        setQuery(person.name);
+                        setIsActive(false);
+                      }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here